fast64_internal/oot/c_writer/oot_scene_table_c.py

```diff
@@ -1,75 +1,144 @@
-import os, re
+import os
 from ...utility import *
+from ..oot_constants import ootEnumSceneID
 
-class SceneTableEntry:
-	def __init__(self, sceneType, name, title, unk10, config, unk12):
-		self.sceneType = sceneType
-		self.name = name
-		self.title = title
-		self.unk10 = unk10
-		self.config = config
-		self.unk12 = unk12
-
-	def toC(self):
-		return "\t" + self.sceneType + "(" + self.name + ", " + \
-			((self.title + ", ") if self.title is not None else "")  +\
-			self.unk10 + ", " + self.config + ", " + self.unk12 + "),\n"
-
-def getSceneTableEntryBySceneName(sceneTable, sceneName):
-	for entry in sceneTable:
-		if entry.name[:-6] == sceneName:
-			return entry
-	return None
-
-def readSceneTable(exportPath):
-	fileData = readFile(os.path.join(exportPath, 'src/code/z_scene_table.c'))
-
-	matchResult = re.search('SceneTableEntry\s*gSceneTable\[\]\s*=\s*\{([^\}]*)\}', fileData, re.DOTALL)
-	if matchResult is None:
-		raise PluginError("z_scene_table.c does not have gSceneTable in it.")
-	sceneTable = parseSceneTableData(matchResult.group(1))
-
-	return sceneTable, fileData, matchResult.start(0), matchResult.end(0)
-
-def parseSceneTableData(data):
-	table = []
-	for match in re.finditer('([^,]*)\s*\(([^,]*),\s*([^,]*),\s*([^,]*),\s*([^,]*)(,\s*([^,]*))?\)\s*,', data):
-		name = match.group(2)
-		sceneType = match.group(1).strip()
-		if sceneType == "UNTITLED_SCENE":
-			table.append(SceneTableEntry(sceneType, name, None, match.group(3), match.group(4), match.group(5)))
-		elif sceneType == "TITLED_SCENE":
-			table.append(SceneTableEntry(sceneType, name, match.group(3), match.group(4), match.group(5), match.group(7)))
-		else:
-			raise PluginError("Unhandled scene entry type: " + str(sceneType))
-	return table
-
-def sceneTableToString(sceneTable):
-	data = 'SceneTableEntry gSceneTable[] = {\n'
-	for entry in sceneTable:
-		data += entry.toC()
-	data += '}'
-	return data
-
-def writeSceneTable(sceneTable, fileData, start, end, exportPath):
-	sceneTableData = sceneTableToString(sceneTable)
-	newFileData = fileData[:start] + sceneTableData + fileData[end:]
-
-	if newFileData != fileData:
-		writeFile(os.path.join(exportPath, 'src/code/z_scene_table.c'), newFileData)
+def getSceneTable(exportPath):
+	'''Read and remove unwanted stuff from ``scene_table.h``'''
+	dataList = []
+	fileHeader = ""
+	debugLine = ""
+
+	# read the scene table
+	try:
+		with open(os.path.join(exportPath, 'include/tables/scene_table.h')) as fileData:
+			# keep the relevant data and do some formatting
+			for i, line in enumerate(fileData):
+				if not line.startswith("// "):
+					if not (line.startswith("/**") or line.startswith(" *")):
+						dataList.append(
+							line[(line.find("(") + 1) :]
+							.rstrip(")\n")
+							.replace(" ", "")
+							.split(",")
+						)
+					else:
+						fileHeader += line
+				elif not line.startswith("// Added scenes"):
+					debugLine = line
+	except:
```

Comment:
```suggestion
	except FileNotFoundError:
```
to avoid catching all kinds of exceptions